jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/AbstractConnector.java

```diff
@@ -321,6 +323,36 @@ public boolean isShutdownDone()
         LOG.info("Started {}", this);
     }
 
+    public ComplianceViolation.Listener getComplianceViolationListener()
+    {
+        if (_complianceViolationListener == null)
+        {
+            Server server = getServer();
+
+            // Only add the ComplianceViolations instance if the recording of Compliance Violations is enabled
+            // This also means that any user provided ComplianceViolation.ListenerFactory beans will only be
+            // used when the configuration on the HttpConnectionFactory allows then to be used.
+            HttpConnectionFactory httpConnectionFactory = getConnectionFactory(HttpConnectionFactory.class);
```

Comment:
Only works for HTTP/1.1, not for HTTP/2 or HTTP/3.
We should move `isRecordHttpComplianceViolations()` to `HttpConfiguration` and rename it more generically to  `isNotifyComplianceViolations()` (because we notify HTTP, cookie, URI and multipart violations).